packages/react-menu/src/components/MenuList/useMenuList.ts

```diff
@@ -12,14 +13,58 @@ export const useMenuList = (
   ref: React.Ref<HTMLElement>,
   defaultProps?: MenuListProps,
 ): MenuListState => {
+  const focusAttributes = useArrowNavigationGroup({ circular: true });
+  const { findAll } = useFocusFinders();
+
   const state = mergeProps(
     {
       ref: useMergedRefs(ref, React.useRef(null)),
       role: 'menu',
+      ...focusAttributes,
     },
     defaultProps,
     resolveShorthandProps(props, []),
   );
 
+  state.setFocusByFirstCharacter = React.useCallback(
+    (e: React.KeyboardEvent<HTMLElement>, itemEl: HTMLElement) => {
+      const menuItems = findAll(
```

Comment:
```suggestion
      const menuItems = findAllFocusable(
```

nit: it seems that we should use better naming as `findAll()` in this contents has no sense 💭 

---

Another issue is that we should consider virtualization there, this will not work well with virtualized items.

Reply:
That's a good call out for the naming.

I totally agree with virtualization, but I'm not completely sure that this should be handled here. In a virtualizes collection, you'd need to scroll first to make what you're looking for visible. 

I'll add a TODO... but might not be able to tell much without implmenting a virtualized example